examples/platform/silabs/SiWx917/SiWx917/hal/rsi_hal_mcu_timer.c

```diff
@@ -257,23 +193,20 @@ uint32_t GetTickCount(void)
  * @param[out]   none
  * @return       none
  * @description  This HAL API should contain the code to create delay in milli seconds
+ *
  */
+
 void rsi_delay_ms(uint32_t delay_ms)
```

Comment:
This function can be simplified and be cleaner
```
// Probably don't need a uint32_t as delay but lets say we keep it.
void rsi_delay_ms(uint32_t delay_ms)
{
    /// uint64 to protect from an overflow.
    uint64_t waitTime = rsi_hal_gettickcount() + delay_ms;
    
    while (waitTime > rsi_hal_gettickcount())
    {
        // we busy wait
    }
}
```
Here I am assuming that each tick (returned by rsi_hal_gettickcount()) = 1 ms which might not be true but you implemented it that way. Please confirm this.

Reply:
updated in latest commit